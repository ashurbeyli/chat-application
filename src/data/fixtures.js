export const message = {
  author: 'Ninja',
  message: 'Hello world',
  timestamp: 1522409668934,
  token: 'sySt29CMMLV4',
  _id: '5abe20c45d4b5600196c67b8'
};

export const message2 = {
  author: 'martin57',
  message: 'Thanks Peter',
  timestamp: 1522409897141,
  token: 'sySt29CMMLV4',
  _id: '5abe21a95d4b5600196c67b9'
};

export const messages = [message, message2];