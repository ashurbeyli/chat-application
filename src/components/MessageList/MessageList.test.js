import React from 'react';
import { shallow } from 'enzyme';

import { MessageList } from './';
import { messages } from '../../data/fixtures';

const props = { messages };

describe('MessageList', () => {
  const wrapper = shallow(<MessageList {...props}/>);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render MessageCard', () => {
    expect(wrapper.find('[user="Tom"]').length).toEqual(props.messages.length);
  });

});