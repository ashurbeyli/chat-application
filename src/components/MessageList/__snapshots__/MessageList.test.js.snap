// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessageList
    messages={
        Array [
            Object {
              "_id": "5abe20c45d4b5600196c67b8",
              "author": "Ninja",
              "message": "Hello world",
              "timestamp": 1522409668934,
              "token": "sySt29CMMLV4",
            },
            Object {
              "_id": "5abe21a95d4b5600196c67b9",
              "author": "martin57",
              "message": "Thanks Peter",
              "timestamp": 1522409897141,
              "token": "sySt29CMMLV4",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          data={
                    Object {
                              "_id": "5abe20c45d4b5600196c67b8",
                              "author": "Ninja",
                              "message": "Hello world",
                              "timestamp": 1522409668934,
                              "token": "sySt29CMMLV4",
                            }
          }
          user="Tom"
/>,
        <Unknown
          data={
                    Object {
                              "_id": "5abe21a95d4b5600196c67b9",
                              "author": "martin57",
                              "message": "Thanks Peter",
                              "timestamp": 1522409897141,
                              "token": "sySt29CMMLV4",
                            }
          }
          user="Tom"
/>,
      ],
      "className": "message-list",
      "id": "messageList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "5abe20c45d4b5600196c67b8",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "_id": "5abe20c45d4b5600196c67b8",
            "author": "Ninja",
            "message": "Hello world",
            "timestamp": 1522409668934,
            "token": "sySt29CMMLV4",
          },
          "user": "Tom",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5abe21a95d4b5600196c67b9",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "_id": "5abe21a95d4b5600196c67b9",
            "author": "martin57",
            "message": "Thanks Peter",
            "timestamp": 1522409897141,
            "token": "sySt29CMMLV4",
          },
          "user": "Tom",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            data={
                        Object {
                                    "_id": "5abe20c45d4b5600196c67b8",
                                    "author": "Ninja",
                                    "message": "Hello world",
                                    "timestamp": 1522409668934,
                                    "token": "sySt29CMMLV4",
                                  }
            }
            user="Tom"
/>,
          <Unknown
            data={
                        Object {
                                    "_id": "5abe21a95d4b5600196c67b9",
                                    "author": "martin57",
                                    "message": "Thanks Peter",
                                    "timestamp": 1522409897141,
                                    "token": "sySt29CMMLV4",
                                  }
            }
            user="Tom"
/>,
        ],
        "className": "message-list",
        "id": "messageList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "5abe20c45d4b5600196c67b8",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "_id": "5abe20c45d4b5600196c67b8",
              "author": "Ninja",
              "message": "Hello world",
              "timestamp": 1522409668934,
              "token": "sySt29CMMLV4",
            },
            "user": "Tom",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5abe21a95d4b5600196c67b9",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "_id": "5abe21a95d4b5600196c67b9",
              "author": "martin57",
              "message": "Thanks Peter",
              "timestamp": 1522409897141,
              "token": "sySt29CMMLV4",
            },
            "user": "Tom",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
